packages/shared/src/utils.ts

```diff
@@ -105,7 +105,14 @@ export function getTypeFromFileName(
 }
 
 export function generateUploadThingURL(path: `/${string}`) {
-  const host = process.env.CUSTOM_INFRA_URL ?? "https://uploadthing.com";
+  let host = "https://uploadthing.com";
+  try {
+    host = process.env.CUSTOM_INFRA_URL ?? host;
+  } catch (e) {
+    // @ts-expect-error - import.meta is dumb
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+    host = import.meta.env.CUSTOM_INFRA_URL ?? host;
+  }
```

Comment:
this is the only lib change we need to make

i tried making this a reusable helper:

![image](https://github.com/pingdotgg/uploadthing/assets/51714798/1b681efa-dd7d-4a8b-ae67-80501c4b4694)

but it didn't work as the vars are statically analyzed so this function is always returning undefined,